taskApp/src/app/web-request.service.ts

```diff
@@ -9,18 +10,15 @@ export class WebRequestService {
   readonly ROOT_URL;
 
   constructor(private http: HttpClient) { 
-    this.ROOT_URL = 'http://root:example@localhost:27017';
+    this.ROOT_URL=environment.apiUrl;
   }
 
   get(url: string){
       return this.http.get(`${this.ROOT_URL}/${url}`);
   }
-  post(url: string, payload: Object){
+  post(url: string, payload: string){
     return this.http.post(`${this.ROOT_URL}/${url}`,payload);
   }
```

Comment:
Just so you know.  Post requests in angular HTTP allow you to send up an object as well, not just a string.  I think this is where your problems were happening before?